python/lsst/meas/extensions/shapeHSM/_hsm_moments.py

```diff
@@ -0,0 +1,243 @@
+# This file is part of meas_extensions_shapeHSM.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import galsim
+import lsst.afw.geom as afwGeom
+import lsst.afw.table as afwTable
+import lsst.meas.base as measBase
+import lsst.pex.config as pexConfig
+from lsst.meas.base import (
+    FlagDefinitionList,
+    FlagHandler,
+    MeasurementError,
+    SingleFramePlugin,
+    SingleFramePluginConfig,
+)
+from lsst.pex.exceptions import InvalidParameterError
+from lsst.geom import Point2I, Point2D
+
+
+class HsmMomentsConfig(SingleFramePluginConfig):
+    roundMoments = pexConfig.Field[bool](doc="Use round weight function?", default=False)
+    addFlux = pexConfig.Field[bool](doc="Store measured flux?", default=False)
+    subtractCenter = pexConfig.Field[bool](doc="Subtract starting center from x/y outputs?", default=False)
+
+
+class HsmMomentsPlugin(SingleFramePlugin):
+    ConfigClass = HsmMomentsConfig
+
+    def __init__(self, config, name, schema, logName=None):
+        super().__init__(config, name, schema, logName)
+
+        # Flag definitions.
+        flagDefs = FlagDefinitionList()
+        self.FAILURE = flagDefs.add("flag", "General failure flag, set if anything went wrong")
```

Comment:
I suspect that this one alone should be `.addFailureFlag` without the first argument, instead of `.add`. We can come back to this later.